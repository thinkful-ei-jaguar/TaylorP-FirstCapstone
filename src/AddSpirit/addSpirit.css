* {
  box-sizing: border-box;
}

.spirit_main {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px 0;
}

.add-spirit-form {
  border: 1px solid #025951;
  background-color: #A6864B;
  padding: 20px;
  width: 300px;
  border-radius: 5px;
  font-weight: 600;
  font-size: 18px;
}

.add-spirit-form input {
  width: 100%;
}

.spirit-select-button-div {
  width: 100%;
  padding-top: 10px; 
}

.spirit-select-button-div label {
  padding-right: 5px;
}

.spirit-select-button-div button {
  float: right;
}

.go-back {
  width: 100px;
  margin-top: 20px;
  padding: 10px;
  border-radius: 5px;
  background-color: #067368;
  color: #DCF2F0;
  font-size: 20px;
  font-weight: 600;
  font-family: 'Josefin Slab', serif;
}