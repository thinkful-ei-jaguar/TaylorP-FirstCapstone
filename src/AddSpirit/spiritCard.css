.spirit-card-li {
  border: 1px solid black;
  background-color: lightgray;
  width: 170px;
  height: 170px;
  display: flex;
  flex-direction: column;
  padding: 5px;
  margin: 10px auto;
  text-align: center;
}

.spirit-card-li div {
  width: 100px;
  height: 100px;
  margin: 0 auto;
  border: 1px dashed grey;
}

.spirit-card-li h2, .spirit-card-li h3 {
  margin: 0;
}