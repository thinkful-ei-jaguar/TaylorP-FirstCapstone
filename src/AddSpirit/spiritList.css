.spirit-list {
  display: flex;
  flex-wrap: wrap;
}