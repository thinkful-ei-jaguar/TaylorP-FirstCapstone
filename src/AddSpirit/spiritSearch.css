* {
  box-sizing: border-box;
}

.spirit-search {
  display: flex;
  justify-content: center;
  padding: 20px 0;
}

.spirit-search-form {
  border: 1px solid black;
  padding: 10px;
  width: 300px;
  border-radius: 5px;
}

.spirit-search-form input {
  width: 100%;
}

.spirit-select-button-div {
  width: 100%;
  padding-top: 10px; 
}

.spirit-select-button-div label {
  padding-right: 5px;
}

.spirit-select-button-div button {
  float: right;
}