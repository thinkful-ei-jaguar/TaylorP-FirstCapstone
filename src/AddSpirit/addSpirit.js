import React, { Component } from 'react'
import SpiritApiService from '../Services/spirit-api-service'
import TokenService from '../Services/token-service'
import SpiritListContext from '../Context/spiritListContext'
import './addSpirit.css'


class AddSpirit extends Component {
  static contextType = SpiritListContext

  handleSpiritSubmit = e => {
    e.preventDefault()
    const { spiritList } = this.context
    const { spirit_name, spirit_id } = e.target
    const user_id = TokenService.getUserId()
    const spirit = {spirit_name:spirit_name.value, spirit_id:Number(spirit_id.value), user_id} 
    //const pushToCabinet = function() { this.props.history.push('/cabinet') }

    SpiritApiService.postSpirit(spirit, user_id)
    .then(this.context.setSpiritList([...spiritList, spirit]))
    .then(this.context.setSpiritAddTrue())
    //.then(setTimeout(this.props.history.push('/cabinet'), 5000))
  }

  goBack = () => {
    this.context.setSpiritAddFalse();
    this.props.history.goBack();
  }

  render(){
    const { spiritAdd } = this.context
    return (
      <main className='spirit_main'>
        <form className='add-spirit-form' onSubmit={this.handleSpiritSubmit}>
          <div className='namebar'>
            <label htmlFor='spirit_name'>Spirit Name:{' '}</label>
            <input type='text' name='spirit_name' id='spirit_name' placeholder='Ex: Espolon' />
          </div>
          <div className='spirit-select-button-div'>
            <label htmlFor='spirit_id'>Spirit:</label>
            <select name='spirit_id' id='spirit_id'>
              <option value='2'>Vodka</option>
              <option value='3'>Gin</option>
              <option value='1'>Tequila</option>
              <option value='4'>Bourbon</option>
              <option value='7'>Rye</option>
              <option value='5'>Scotch</option>
              <option value='6'>Amaro</option>
              <option value='8'>Cognac</option>
              <option value='9'>Rum</option>
            </select>
            <button type='submit'>
              Save
            </button>
          </div>
        </form>
        {spiritAdd && <p className='spirit-add-alert'>Spirit Added to your Cabinet</p>}
        <button type='button' className='go-back' onClick={this.goBack} >Go Back</button>
      </main>
    )
  }
}

export default AddSpirit;