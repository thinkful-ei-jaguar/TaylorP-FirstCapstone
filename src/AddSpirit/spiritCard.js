import React, { Component } from 'react'
import './spiritCard.css'

class SpiritCard extends Component {
  render(){
    return (
      <li className='spirit-card-li'>
        <div>img</div>
        <h2>Espolon</h2>
        <h3>Tequila</h3>
      </li>
    )
  }
}

export default SpiritCard;