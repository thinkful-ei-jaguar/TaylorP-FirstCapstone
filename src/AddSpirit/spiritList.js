import React, { Component } from 'react'
import SpiritCard from './spiritCard'
import './spiritList.css'


class SpiritList extends Component {
  render(){
    return (
      <ul className='spirit-list' aria-live='polite'>
        <SpiritCard />
        <SpiritCard />
        <SpiritCard />
        <SpiritCard />
        <SpiritCard />
        <SpiritCard />
      </ul>
    )
  }
}

export default SpiritList;