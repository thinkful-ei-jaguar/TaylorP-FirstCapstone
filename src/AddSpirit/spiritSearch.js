import React, {Component} from 'react'
import './spiritSearch.css'


class SpiritSearch extends Component {
  render(){
    return (
      <section className='spirit-search'>
        
      </section>
    )
  }
}

export default SpiritSearch;