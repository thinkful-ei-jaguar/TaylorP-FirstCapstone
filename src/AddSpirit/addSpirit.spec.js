import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import toJson from 'enzyme-to-json'
import AddSpirit from './addSpirit'
import RecipeContext from '../Context/recipeContext'

describe('<AddSpirit />', () => {
  it('renders without crashing', () => {
    const match = {params: {id: 1}}
    mount(
      <MemoryRouter>
        <AddSpirit match={match}/>
      </MemoryRouter>
    )
  })
  it('renders as expected', () => {
    const match = { params: { id: 1 } }
    const tree = renderer.create(
      <MemoryRouter>
        <AddSpirit match={match}/>
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot();
  })
})