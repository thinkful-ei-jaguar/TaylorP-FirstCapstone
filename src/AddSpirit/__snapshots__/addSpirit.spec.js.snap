// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddSpirit /> renders as expected 1`] = `
<main
  className="spirit_main"
>
  <form
    className="add-spirit-form"
    onSubmit={[Function]}
  >
    <div
      className="namebar"
    >
      <label
        htmlFor="spirit_name"
      >
        Spirit Name:
         
      </label>
      <input
        id="spirit_name"
        name="spirit_name"
        placeholder="Ex: Espolon"
        type="text"
      />
    </div>
    <div
      className="spirit-select-button-div"
    >
      <label
        htmlFor="spirit_id"
      >
        Spirit:
      </label>
      <select
        id="spirit_id"
        name="spirit_id"
      >
        <option
          value="2"
        >
          Vodka
        </option>
        <option
          value="3"
        >
          Gin
        </option>
        <option
          value="1"
        >
          Tequila
        </option>
        <option
          value="4"
        >
          Bourbon
        </option>
        <option
          value="7"
        >
          Rye
        </option>
        <option
          value="5"
        >
          Scotch
        </option>
        <option
          value="6"
        >
          Amaro
        </option>
        <option
          value="8"
        >
          Cognac
        </option>
        <option
          value="9"
        >
          Rum
        </option>
      </select>
      <button
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
  <button
    className="go-back"
    onClick={[Function]}
    type="button"
  >
    Go Back
  </button>
</main>
`;
